### PR TITLE
requirements: use django-cors-headers instead

### PR DESCRIPTION
**Pull request by humitos**
Note that `django-cors-middleware` is deprecated (see note in
https://github.com/zestedesavoir/django-cors-middleware) and they recommend
using `django-cors-headers` instead.

Related to #8692

### CODE REVIEW

File: requirements/pip.txt [hunk 1]
```diff
@@ -92,9 +92,7 @@ dj-pagination==2.5.0
 # Version comparison stuff
 packaging==21.2
 
-# django-cors-middleware==1.5.0 fails with
-# AttributeError: 'dict' object has no attribute 'has_header'
-django-cors-middleware==1.4.0  # pyup: ignore
+django-cors-headers==3.10.0
```
Comment (humitos) [inline requirements/pip.txt:95]:
```suggestion
django-cors-headers==2.0.0
```

File: requirements/pip.txt [hunk 2]
```diff
@@ -92,9 +92,7 @@ dj-pagination==2.5.0
 # Version comparison stuff
 packaging==21.2
 
-# django-cors-middleware==1.5.0 fails with
-# AttributeError: 'dict' object has no attribute 'has_header'
-django-cors-middleware==1.4.0  # pyup: ignore
+django-cors-headers==2.0.0
```
Comment (humitos) [inline requirements/pip.txt:95]:
```suggestion
# django-cors-middleware was replaced by django-cors-headers
# django-cors-headers==3.10.0 is failing with:
# AttributeError: 'dict' object has no attribute 'has_header'
django-cors-headers==2.0.0  # pyup: ignore
```

File: requirements/pip.txt [hunk 3]
```diff
@@ -92,9 +92,10 @@ dj-pagination==2.5.0
 # Version comparison stuff
 packaging==21.2
 
-# django-cors-middleware==1.5.0 fails with
+# django-cors-middleware was replaced by django-cors-headers
+# django-cors-headers==3.10.0 is failing with:
```
Comment (ericholscher) [inline requirements/pip.txt:96]: Should we not fix the error if we're looking to upgrade this package? This feels like a sideways move, but not really what we need to do to move forward?

**Comment by agjohnson** [comment]
Bumping this one off our release tomorrow due to questions here.

File: requirements/pip.txt (hunk 3 above)
Comment (humitos) [reply requirements/pip.txt:96]: I don't think we can fix this issue by ourselves. I didn't research too much yet, but I found https://github.com/adamchainz/django-cors-headers/issues/558 that it's related to a Django bug which is solved in 3.1.x

File: requirements/pip.txt [hunk 4]
```diff
@@ -92,9 +92,10 @@ dj-pagination==2.5.0
...
 # AttributeError: 'dict' object has no attribute 'has_header'
-django-cors-middleware==1.4.0  # pyup: ignore
+django-cors-headers==2.0.0  # pyup: ignore
```
Comment (humitos) [inline requirements/pip.txt:98]:
```suggestion
django-cors-headers==3.11.0
```

**Comment by agjohnson** [comment]
This seems like it will get bumped out a release. I won't reassign a release yet though, sounds like this one is a bit more elusive than we expected.

File: requirements/pip.txt (hunk 3 above)
Comment (ericholscher) [reply requirements/pip.txt]: @humitos I believe we're on a newer Django now, so is this fixed?

Comment (humitos) [reply requirements/pip.txt]: Let's see. I've rebased the branch and push. Tests should pass now if it's fixed 😄

Comment (humitos) [reply requirements/pip.txt]: It seems it's not fixed yet... Tests are still failing, :/

**Comment by humitos** [comment]
With the latest refactor that @stsewd did in CORS, this dependency can probably be removed completely, right?

**Comment by stsewd** [comment]
@humitos nope, we still need it to set the proper CORS headers for us.

**Comment by ericholscher** [comment]
@stsewd would be good to get upgraded to a supported version here, since we aren't doing anything custom anymore. 👍

**Comment by stsewd** [comment]
I'm doing this at https://github.com/readthedocs/readthedocs.org/pull/10000.

also, issue number 10K!

**Pull request closed without merging**